instrumentation/guava-10.0/src/main/java/io/opentelemetry/instrumentation/auto/guava/ListenableFutureInstrumentation.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package datadog.trace.instrumentation.guava10;
+
+import static java.util.Collections.singletonMap;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+import com.google.auto.service.AutoService;
+import com.google.common.util.concurrent.AbstractFuture;
+import io.grpc.Context;
+import io.opentelemetry.instrumentation.auto.api.ContextStore;
+import io.opentelemetry.instrumentation.auto.api.InstrumentationContext;
+import io.opentelemetry.instrumentation.auto.api.concurrent.ExecutorInstrumentationUtils;
+import io.opentelemetry.instrumentation.auto.api.concurrent.RunnableWrapper;
+import io.opentelemetry.instrumentation.auto.api.concurrent.State;
+import io.opentelemetry.javaagent.tooling.Instrumenter;
+import java.util.Map;
+import java.util.concurrent.Executor;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+
+@AutoService(Instrumenter.class)
+public class ListenableFutureInstrumentation extends Instrumenter.Default {
+
+  public ListenableFutureInstrumentation() {
+    super("guava");
+  }
+
+  @Override
+  public ElementMatcher<TypeDescription> typeMatcher() {
+    return named("com.google.common.util.concurrent.AbstractFuture");
+  }
+
+  @Override
+  public Map<String, String> contextStore() {
+    return singletonMap(Runnable.class.getName(), State.class.getName());
+  }
+
+  @Override
+  public Map<? extends ElementMatcher<? super MethodDescription>, String> transformers() {
+    return singletonMap(
+        named("addListener").and(ElementMatchers.takesArguments(Runnable.class, Executor.class)),
+        ListenableFutureInstrumentation.class.getName() + "$AddListenerAdvice");
+  }
+
+  public static class AddListenerAdvice {
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    public static State addListenerEnter(
+        @Advice.Argument(value = 0, readOnly = false) Runnable task,
+        @Advice.Argument(1) final Executor executor) {
+      final Context context = Context.current();
+      final Runnable newTask = RunnableWrapper.wrapIfNeeded(task);
+      // It is important to check potentially wrapped task if we can instrument task in this
+      // executor. Some executors do not support wrapped tasks.
```

Comment:
@trask You have removed this check in another PR, haven't you?

Reply:
yes, thanks, and this comment prompted #1128